### ISSUE TITLE
Training Convergence

### ISSUE THREAD

**Issue by iamweiweishi** [microsoft/INMT-lite#8]
Hi, 
Thank you for your great work.
I followed your instructions and tried to train a TFLite model.
But it looks like the models do not converge well.

The loss is as follows:
Epoch 1 Batch 0 Loss 10.1416
Epoch 1 Batch 100 Loss 6.4002
Epoch 1 Batch 200 Loss 5.1489
Epoch 1 Training Loss 6.0853
Time taken for 1 epoch 76.00192999839783 sec

Epoch 1 Validation Loss 11.4574
Time taken for validation 2.3923656940460205 sec

......

Epoch 99 Batch 0 Loss 1.8126
Epoch 99 Batch 100 Loss 2.3800
Epoch 99 Batch 200 Loss 2.4746
Epoch 99 Training Loss 2.4855
Time taken for 1 epoch 52.326969385147095 sec

Epoch 99 Validation Loss 25.6011
Time taken for validation 2.445949077606201 sec

Epoch 100 Batch 0 Loss 2.5111
Epoch 100 Batch 100 Loss 2.7126
Epoch 100 Batch 200 Loss 2.5386
Epoch 100 Training Loss 2.5118
Time taken for 1 epoch 56.2775776386261 sec

Epoch 100 Validation Loss 25.9406
Time taken for validation 6.213912010192871 sec

And I tested the ACC of the model. It is only around 8.3.

What should I do to improve the performance of the model.

**Comment by anuragshukla06** [comment]
Hi!
You may want to increase the number of hidden units of the model, the default is 500. It can be done by passing `--recurrent_hidden <number of hidden units>` flag to `train.py`, for Eg, `--recurrent_hidden 1024`. Also, if your vocabulary is large, you may try to increase the number of embedding layer units, the default is 256 for both encoder and decoder. It can be changed by passing `--src_word_vec_size <vector_size>` to change the encoder's embedding size and `--tgt_word_vec_size` <vector_size> to change the decoder's embedding size.
